src/d/parser/expression.d

```diff
@@ -1043,3 +1046,80 @@ StringLiteral parseStringLiteral(ref TokenRange trange) {
 	
 	return new StringLiteral(location, name.toString(trange.context));
 }
+
+/**
+ * Parse floating point literals
+ */
+FloatLiteral parseFloatLiteral(ref TokenRange trange) {
+	const location = trange.front.location;
+	auto litString = trange.front.toString(trange.context);
+
+	trange.match(TokenType.FloatLiteral);
+	import d.common.builtintype : BuiltinType;
+
+
+	assert(litString.length > 1);
+	//Look for a suffix
+	switch (litString[$-1])
+	{
+		// https://dlang.org/spec/lex.html#FloatSuffix
+		import std.conv : to;
+		case 'f':
+		case 'F':
+			const float f = litString[0..$-1].to!float;
+			return new FloatLiteral(location, f, BuiltinType.Float);
+		case 'L':
+			import source.exception;
+			throw new CompileException(location, "SDC does not support real literals yet");
+		default:
+			//Lexed correctly but no suffix, it's a double
+			const double d = litString[0..$].to!double;
+			return new FloatLiteral(location, d, BuiltinType.Double);
+	}
+}
+
+@("Test FloatLiteral parsing")
+unittest
+{
+	import source.context;
+	auto context = new Context();
+	import source.parserutil;
+	auto tokensFromString(string s) {
+		import source.name;
+		auto base = context.registerMixin(Location.init, s ~ '\0');
+		auto x = lex(base, context);
+		x.match(TokenType.Begin);
+		return x;
+	}
+	void floatRoundTrip(T)(const string floatString, const T floatValue)
+		if (__traits(isFloating, T))
+	{
+		import d.common.builtintype : BuiltinType;
+		const BuiltinType expectedType = is(T == float) ? BuiltinType.Float : BuiltinType.Double;
+		//Acceptable relativeError
+		const auto maxRelError = is(T == float) ? float.epsilon : double.epsilon;
+		auto tr = tokensFromString(floatString);
+		const result = parseFloatLiteral(tr);
+		import std.format : format;
+		const FloatLiteral fl = cast(FloatLiteral) result;
```

Comment:
Why do you need to cast?

Reply:
Well that's leftover from when it used to be parseExpression i.e. to make sure the result is a real FloatLiteral. I'll get rid